### DISCUSSION TITLE
Log hyper-parameters in logger

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Lightning-AI/pytorch-lightning#12273] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
i am passing different hyper-parameters to my model and i want to log the output using logger. But in my case it create a new logger for each parameter. is there a way to have one logger and create multiple graph on it.
Here is the code where i want to check which scheduler work the best. 
```
class OurModel(LightningModule):
    def __init__(self,scheduler):
        super(OurModel,self).__init__()

        self.scheduler=scheduler
        self.model=CNNLSTM()
    def forward(self,x):
        return self.model(x)

    def configure_optimizers(self):
        opt=torch.optim.AdamW(params=self.parameters(),lr=self.lr )
        if self.scheduler=='cosine':
            scheduler=CosineAnnealingLR(opt,T_max=10,  eta_min=1e-6, last_epoch=-1)
            return {'optimizer': opt,'lr_scheduler':scheduler}
        elif self.scheduler=='reduce':
            scheduler=ReduceLROnPlateau(opt,mode='min', factor=0.5, patience=5)
            return {'optimizer': opt,'lr_scheduler':scheduler,'monitor':'val_loss'}
        elif self.scheduler=='warm':
            scheduler=CosineAnnealingWarmRestarts(opt,T_0=10, T_mult=1, eta_min=1e-6, last_epoch=-1)
            return {'optimizer': opt,'lr_scheduler':scheduler}
        elif self.scheduler=='cycle':
            opt=torch.optim.AdamW(params=self.parameters(),lr=1e-6 )
            scheduler=OneCycleLR(opt,max_lr=1e-2,epochs=15,steps_per_epoch=len(self.train_df)//self.batch_size//4)
            lr_scheduler = {'scheduler': scheduler, 'interval': 'step'}
            return {'optimizer': opt, 'lr_scheduler': lr_scheduler}
        elif self.scheduler=='lambda':
            lambda1 = lambda epoch: 0.9 ** epoch
            scheduler = torch.optim.lr_scheduler.LambdaLR(opt, lr_lambda=lambda1)
            return {'optimizer': opt, 'lr_scheduler': scheduler}
        elif self.scheduler=='constant':
            return opt
        
    def train_dataloader(self):
        train_dataset=..............
        train_loader=DataLoader(train_dataset,.............)
        return train_loader

    def training_step(self,batch,batch_idx):
        ...................................
        return {'loss':loss,'mae':mae.detach()}

    def training_epoch_end(self, outputs):
        .................................
        self.log('train_loss', loss,batch_size=self.batch_size)
        self.log('train_mae', mae,batch_size=self.batch_size)
        print('training loss ',self.current_epoch,loss,mae)
   
for scheduler in ['cosine','warm','constant','lambda']:
    model=OurModel(scheduler)
    neptune_logger = NeptuneLogger(
                    api_key=api_token,
                    project="",)
    trainer = Trainer(
            logger=neptune_logger )
    trainer.fit(model)

```

**Comment by akihironitta** [comment]
Hi @talhaanwarch!

Call `self.save_hyperparameters()` in your `LightningModule`, and you'll see their logged values (in `HPARAMS` tab if you're using tensorboard).
```diff
class OurModel(LightningModule):
    def __init__(self,scheduler):
        super(OurModel,self).__init__()
+       self.save_hyperparameters()

        self.scheduler=scheduler
        self.model=CNNLSTM()
```

https://pytorch-lightning.readthedocs.io/en/1.5.10/common/lightning_module.html#save-hyperparameters
